### PR TITLE
Allow to connect to ES cluster using HTTPS

### PR DESCRIPTION
**Pull request by beowulfenator**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | Supercedes PR #94 (I hope)

### CODE REVIEW

File: Connection.php [hunk 1]
```diff
@@ -151,22 +169,34 @@ protected function populateNodes()
     {
         $node = reset($this->nodes);
         $host = $node['http_address'];
+        $protocol = isset($node['protocol']) ? $node['protocol'] : $this->defaultProtocol;
         if (strncmp($host, 'inet[/', 6) === 0) {
             $host = substr($host, 6, -1);
         }
-        $response = $this->httpRequest('GET', 'http://' . $host . '/_nodes');
+        $response = $this->httpRequest('GET', $protocol.'://' . $host . '/_nodes');
```
Comment (samdark) [inline Connection.php]: Formatting should be corrected. `$protocol . '://'`

File: Connection.php [hunk 2]
```diff
@@ -440,15 +474,15 @@ protected function httpRequest($method, $url, $requestBody = null, $raw = false)
...
+            list($protocol, $host, $q) = $url;
             if (strncmp($host, 'inet[', 5) == 0) {
                 $host = substr($host, 5, -1);
                 if (($pos = strpos($host, '/')) !== false) {
                     $host = substr($host, $pos + 1);
                 }
             }
             $profile = $method . ' ' . $q . '#' . $requestBody;
-            $url = 'http://' . $host . '/' . $q;
+            $url = $protocol.'://' . $host . '/' . $q;
```
Comment (samdark) [inline Connection.php]: Formatting should be fixed.

Comment (samdark) [inline Connection.php]: Yeah.

**Comment by beowulfenator** [comment]
Anything else before I merge this?

**Comment by samdark** [comment]
Nothing about code style or optimizations from me. If it works and makes sense — merge it.

**Comment by hofrob** [comment]
Is this extension deprecated? Or why is this not released yet??

**Comment by cebe** [comment]
going to make a release soon.

**Comment by hofrob** [comment]
I'd seriously consider deprecating this extension and recommending [elastic/elasticsearch-php](https://github.com/elastic/elasticsearch-php). It's quite easy to implement as a Yii component:

```php
use Elasticsearch\ClientBuilder;

return [
    'components' => [
        'elastic' => function () {
            $host = [
                'host'   => 'example.com',
                'port'   => 443,
                'scheme' => 'https',
                'user'   => 'aaa',
                'pass'   => 'bbb',
            ];

            $clientBuilder = ClientBuilder::create();
            $clientBuilder->setHosts([$host]);

            return $clientBuilder->build();
        },
    ],
];
```

or as in my case, if you want to attach your loggers you'll need to wait for the log component to be initialized:

```php
use Elasticsearch\ClientBuilder;

return [
    'components' => [
        'elastic' => function () {
            $host = [
                'host'   => 'example.com',
                'port'   => 443,
                'scheme' => 'https',
                'user'   => 'aaa',
                'pass'   => 'bbb',
            ];

            return function () use ($host) {
                $clientBuilder = ClientBuilder::create();
                $clientBuilder->setHosts([$host]);

                foreach ($loggers as $logger) {
                    $clientBuilder->setLogger($logger);
                }

                return $clientBuilder->build();
            };
        },
    ],
];
```

**Pull request merged**